lib/diagnostic.rb

```diff
@@ -43,6 +43,9 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open('filename') do |file|
+      file.each do |line|
+    puts line
```

Comment:
You're so close! You have to use `.chomp` and + ' ' to alter the lines in the file.